ntoskrnl/cache/section/swapout.c

```diff
@@ -254,7 +254,7 @@ MmFinalizeSectionPageOut(PMM_SECTION_SEGMENT Segment,
     }
 
     /* Note: Writing may evict the segment... Nothing is guaranteed from here down */
-    MiSetPageEvent(Segment, FileOffset->LowPart);
+    MiSetPageEvent(Segment, (ULONG_PTR)FileOffset->QuadPart);
```

Comment:
So if I understand correctly, when compiled for x86 32-bit, the QuadPart would be truncated?

Reply:
yes

Reply:
Is it intentional? (aka. is it expected that on 32-bit, the FileOffset will always hold on 32-bit (which is wrong I think) and/or the rest of the FileOffset will actually be taken into account in some way or another?